File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/database/witnessing/PendingDao.java
package com.github.dedis.popstellar.repository.database.witnessing;

import androidx.room.*;

import com.github.dedis.popstellar.model.objects.security.MessageID;

import io.reactivex.Completable;
import io.reactivex.Single;

@Dao
public interface PendingDao {
  @Insert(onConflict = OnConflictStrategy.REPLACE)
  Completable insert(PendingEntity pendingEntity);

  @Query("SELECT * FROM pending_objects WHERE id = :messageID")
  Single<PendingEntity> getPendingObject(MessageID messageID);

  @Query("DELETE FROM pending_objects WHERE id = :messageID")
  Completable removePendingObject(MessageID messageID);
}

File: fe2-android/app/src/main/java/com/github/dedis/popstellar/repository/database/witnessing/PendingDao.kt
package com.github.dedis.popstellar.repository.database.witnessing

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import com.github.dedis.popstellar.model.objects.security.MessageID
import io.reactivex.Completable
import io.reactivex.Single

@Dao
interface PendingDao {
  @Insert(onConflict = OnConflictStrategy.REPLACE)
  fun insert(pendingEntity: PendingEntity): Completable

  @Query("SELECT * FROM pending_objects WHERE lao_id = :laoId")
  fun getPendingObjectsFromLao(laoId: String): Single<List<PendingEntity>?>

  @Query("DELETE FROM pending_objects WHERE id = :messageID")
  fun removePendingObject(messageID: MessageID): Completable
}
